### PR TITLE
Suppressed ClosedWatchServiceException and updated Java of Archetypes to 11 #188

### PR DESCRIPTION
**Pull request by jiakuan**
Fixes for #188

### CODE REVIEW

**Comment by niloc132** [comment]
CI is giving us some grief, usually it is closer to 5% failure rate (something in the resource constrained env seems to make it easy to hit a race condition when evaluating unit tests in a sample maven project...). I'll look more closely at the output here when I get the chance and confirm we're not actually hitting a real bug of some kind, then get this PR merged.

File: build-caching/src/main/java/com/vertispan/j2cl/build/DiskCache.java [hunk 1]
```diff
@@ -170,6 +170,10 @@ private void checkForWork() {
         } catch (IOException e) {
             // disaster, can't interact with the cache, stop and give up
             // TODO mark all pending work as canceled?
+        } catch (ClosedWatchServiceException e) {
+            // This is purely noise in the log and doesn't indicate an actual
```
Comment (treblereel) [inline build-caching/src/main/java/com/vertispan/j2cl/build/DiskCache.java:174]:
```suggestion
            if(!livenessThread.getState().equals(Thread.State.TERMINATED)) {
                 throw new Error(e);
            }

            // This is purely noise in the log and doesn't indicate an actual
```

Comment (treblereel) [reply build-caching/src/main/java/com/vertispan/j2cl/build/DiskCache.java:174]: I think that we can add an additional check to ensure that this exception is thrown only when the build is complete. This is necessary to rule out other scenarios.

**Review by treblereel — CHANGES_REQUESTED** [review]

**Comment by treblereel** [comment]
> CI is giving us some grief, usually it is closer to 5% failure rate (something in the resource constrained env seems to make it easy to hit a race condition when evaluating unit tests in a sample maven project...). I'll look more closely at the output here when I get the chance and confirm we're not actually hitting a real bug of some kind, then get this PR merged.

in my case, on osx each build ends with this exception.

**Review by treblereel — APPROVED** [review]
It's good to go.
Thanks!

**Pull request merged**